### PR TITLE
add initial content for release doc

### PR DESCRIPTION
**Pull request by elmiko**

### CODE REVIEW

File: docs/release/index.md [hunk 1]
```diff
@@ -1,3 +1,102 @@
...
+
+While new minor versions of OpenShift are released every 4 months, "Z-Stream" releases
+are created much more frequently to address security and critical bug fixes.
+A z-stream release is designated by an increase in the patch level version of the
+release. For example, version "4.8.1" of OpenShift would designate that the current
+patch level is "1", if a new z-stream release is needed it will be version "4.8.2".
+
+In general, backports to older versions of OpenShift will be evaluated on a case-by-case
+basis. Security and critical bug fixes will be considered the highest priority
+for backporting, while features are rarely considered for backport.
```
Comment (lobziik) [inline docs/release/index.md:22]: Maybe better to emphasise that feature backport is really exceptional thing?

**Review by JoelSpeed — COMMENTED** [review]
LGTM

File: docs/release/index.md [hunk 2]
```diff
@@ -1,3 +1,102 @@
...
+piece necessary for being included in an OpenShift release from Red Hat are
+full end-to-end integration of your infrastructure platform into the continuous
+integration(CI) process (this is covered in greater detail in
+[Continuous Integration and Testing][ci]).
+
+When the end-to-end integration tests of the new infrastructure platform are passing
+regularly, the new platform can be included in an OpenShift release. The new
+components will have already been included in nightly and CI payloads, and they
+will now be promoted to the release payload.
+
```
Comment (lobziik) [inline docs/release/index.md:37]: I'm thinking about common block which gives a hight-level overview on Release process.
Smth like
```
Openshift release process consists of several important phases
Feature development -> Feature Freeze -> Code Freeze -> GA

It's important to understand this phases in order to be able to align provider development process with Openshift releases.
```
WDYT?

Comment (lobziik) [reply docs/release/index.md:37]: Probably include `Code Freeze`, `Feature Freeze` and `GA` blocks into this as sub-paragraphs

**Comment by elmiko** [comment]
@lobziik i tried to clean things up based on your suggestions, wdyt?

**Comment by lobziik** [comment]
Yep, great! Thanks a lot @elmiko!
LGTM

**Pull request merged**